libs/feature/editor/src/lib/components/record-form/form-field/form-field-map-container/form-field-map-container.component.ts

```diff
@@ -0,0 +1,171 @@
+      placeKeyword?: Keyword
+      spatialExtents?: DatasetSpatialExtent
+    }
+  }[]
+
+  error = ''
+  mapContext$: Observable<MapContextModel> = this.mapFacade.layers$.pipe(
+    switchMap((layers) =>
+      from(this.mapUtils.getLayerExtent(layers[0])).pipe(
+        catchError(() => {
+          this.error = 'The layer has no extent'
+          return of(undefined)
+        }),
+        map((extent) => {
+          return {
+            layers: [DEFAULT_BASELAYER_CONTEXT, ...layers],
+            view: {
+              extent: extent,
+            },
+          } as MapContextModel
+        })
+      )
+    )
+  )
+
+  mapConfig: MapConfig = getOptionalMapConfig()
+
+  constructor(
+    private mapFacade: MapFacade,
+    private mapUtils: MapUtilsService,
+    private styleService: MapStyleService
+  ) {
+    const fill = new Fill({
+      color: 'transparent',
+    })
+    const stroke = new Stroke({
+      color: 'black',
+      width: 2,
+    })
+    const styles = [
+      new Style({
+        fill: fill,
+        stroke: stroke,
+      }),
+    ]
+    this.styleService.styles.default = this.styleService.createStyleFunction({
+      ...this.styleService.createGeometryStyles({ color: 'black' }),
+      polygon: styles,
+    })
+  }
+
+  ngOnChanges(): void {
+    this.mapFacade.removeLayer(0)
+    if (this.keywordsWithSpatialExtents) {
+      const featureCollection: GeoJSONFeatureCollection = {
+        type: 'FeatureCollection',
+        features: [],
+      }
+
+      Object.keys(this.keywordsWithSpatialExtents).forEach((key) => {
+        if (
+          this.keywordsWithSpatialExtents[key].spatialExtents?.geometries
+            ?.length >= 0
+        ) {
+          this.keywordsWithSpatialExtents[
+            key
+          ].spatialExtents.geometries.forEach((geometry) =>
+            featureCollection.features.push({
+              type: 'Feature',
+              properties: { description: key },
+              geometry: geometry,
+            })
+          )
+        } else if (
+          this.keywordsWithSpatialExtents[key].spatialExtents?.bbox?.length >= 0
+        ) {
+          featureCollection.features.push({
+            type: 'Feature',
+            properties: { description: key },
+            geometry: this.bboxCoordsToGeometry(
+              this.keywordsWithSpatialExtents[key].spatialExtents.bbox
+            ),
+          })
+        }
+      })
+      this.mapFacade.addLayer({
+        type: MapContextLayerTypeEnum.GEOJSON,
+        data: featureCollection,
+        title: 'Spatial extents',
+      })
+    }
+  }
+  bboxCoordsToGeometry(bbox: [number, number, number, number]): Geometry {
+    const geometry = new Polygon([
+      [
+        [bbox[2], bbox[3]],
+        [bbox[2], bbox[1]],
+        [bbox[0], bbox[1]],
+        [bbox[0], bbox[3]],
+        [bbox[2], bbox[3]],
+      ],
+    ])
+
+    const geoJSONGeom = new GeoJSON().writeGeometryObject(geometry)
+    return geoJSONGeom
+  }
+
+  addToMap(
```

Comment:
Where is this used? The name doesn't really match the content of the function, and I couldn't find any reference.

Reply:
Not used anymore